### PR TITLE
docs(ci): add reviewdog examples

### LINKED ISSUE #1046: Document rdjson usage
Since we added reviewdog rdjson output in [#1026](https://github.com/Bearer/bearer/issues/1026) we need to add documenation on the ways it can be used with reviewdog

### PR DESCRIPTION
**Pull request by gotbadger**
## Description
Adds documentation on PR/MR Integration using new Reviewdog compatible output.

## Related
- #1026
- closes #1046

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [x] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

**Comment by gotbadger** [comment]
@markmichon wasn't sure how you handle screenshots let me know if theres some way you want me to do them or if they need compression etc.

**Review by cfabianski — APPROVED** [review]

File: docs/guides/ci-setup.md [hunk 1]
```diff
@@ -48,17 +48,39 @@ image:
...
+
+  artifacts:
+    reports:
       sast: gl-sast-report.json
 ```
 
 These changes set the format to `gitlab-sast` and write an artifact that GitLab can use. Once run, the results of the security scan will display in the Security and Compliance section of the repository.
 
+### Gitlab Merge Request Comments
+Bearer CLI supports [Reviewdog](https://github.com/reviewdog/reviewdog) rdjson format so, you can get direct feedback on your merge requests.
```
Comment (markmichon) [inline docs/guides/ci-setup.md]:
```suggestion
Bearer CLI supports [Reviewdog](https://github.com/reviewdog/reviewdog) rdjson format so you can get direct feedback on your merge requests.
```

File: docs/guides/ci-setup.md [hunk 2]
```diff
@@ -48,17 +48,39 @@ image:
...
+    SHA: $CI_COMMIT_SHA
+    CURRENT_BRANCH: $CI_COMMIT_REF_NAME
+    DEFAULT_BRANCH: $CI_DEFAULT_BRANCH
+  script:
+    - curl -sfL https://raw.githubusercontent.com/Bearer/bearer/main/contrib/install.sh | sh -s -- -b /usr/local/bin
+    - curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b /usr/local/bin
+    - bearer scan . --format=rdjson --output=rd.json
+    - cat rd.json | reviewdog -f=rdjson -reporter=gitlab-mr-discussion
+```
+[Don't forget](https://github.com/reviewdog/reviewdog#reporter-gitlab-mergerequest-discussions--reportergitlab-mr-discussion) to set `REVIEWDOG_GITLAB_API_TOKEN` in your project environment varaibles with a personal API access token.
```
Comment (markmichon) [inline docs/guides/ci-setup.md]:
```suggestion
[Don't forget](https://github.com/reviewdog/reviewdog#reporter-gitlab-mergerequest-discussions--reportergitlab-mr-discussion) to set `REVIEWDOG_GITLAB_API_TOKEN` in your project environment variables with a personal API access token.
```

File: docs/guides/github-action.njk [hunk 3]
```diff
@@ -122,6 +122,44 @@ jobs:
 
 By setting the format and output path, and adding a new upload step, the action will upload SARIF-formatted findings to GitHub's code scanner.
 
+## Code Review Comments
+Bearer CLI supports [Reviewdog](https://github.com/reviewdog/reviewdog) rdjson format so, using any of the reviewdog reporters you can quickly add bearer feedback direct to your pull requests.
```
Comment (markmichon) [inline docs/guides/github-action.njk]:
```suggestion
Bearer CLI supports [Reviewdog](https://github.com/reviewdog/reviewdog) rdjson format so you can use any of the reviewdog reporters to quickly add bearer feedback directly to your pull requests.
```

File: docs/guides/github-action.njk [hunk 4]
```diff
@@ -122,6 +122,44 @@ jobs:
...
+      - uses: reviewdog/action-setup@v1
+        with:
+          reviewdog_version: latest
+      - name: Run reviewdog
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.REVIEWDOG_GITHUB_API_TOKEN }}
+        run: |
+          cat rd.json | reviewdog -f=rdjson -reporter=github-pr-review
+```
+Remeber to setup a personal access [token](https://github.com/settings/personal-access-tokens/new) with read and write permissions and save it to your repo secrets as `REVIEWDOG_GITHUB_API_TOKEN` and you should be good to go.
```
Comment (markmichon) [inline docs/guides/github-action.njk]:
```suggestion
Remember to set a personal access [token](https://github.com/settings/personal-access-tokens/new) with read and write permissions and save it to your repo secrets as `REVIEWDOG_GITHUB_API_TOKEN` and you should be good to go.
```

**Review by markmichon — APPROVED** [review]

**Pull request merged**